tests/test_acl.py

```diff
@@ -779,3 +779,105 @@ def test_V6AclTableDeletion(self, dvs):
         keys = atbl.getKeys()
         # only the default table was left
         assert len(keys) == 1
+
+    def test_RulesWithDiffMaskLengths(self, dvs):
+        db = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+
+        bind_ports = ["Ethernet0", "Ethernet4"]
+        # create ACL_TABLE in config db
+        tbl = swsscommon.Table(db, "ACL_TABLE")
+        fvs = swsscommon.FieldValuePairs([("policy_desc", "test"), ("type", "L3"), ("ports", ",".join(bind_ports))])
+        tbl.set("test_subnet", fvs)
+
+        time.sleep(2)
+
+        subnet_mask_rules = 0
+        #create ACL rules
+        tbl = swsscommon.Table(db, "ACL_RULE")
+        rules = [ [("PRIORITY", "10"), ("PACKET_ACTION", "FORWARD"), ("SRC_IP", "23.103.0.0/18")],
+                  [("PRIORITY", "20"), ("PACKET_ACTION", "FORWARD"), ("SRC_IP", "104.44.94.0/23")],
+                  [("PRIORITY", "30"), ("PACKET_ACTION", "FORWARD"), ("DST_IP", "172.16.0.0/12")],
+                  [("PRIORITY", "40"), ("PACKET_ACTION", "FORWARD"), ("DST_IP", "100.64.0.0/10")],
+                  [("PRIORITY", "50"), ("PACKET_ACTION", "FORWARD"), ("DST_IP", "104.146.32.0/19")],
+                  [("PRIORITY", "60"), ("PACKET_ACTION", "FORWARD"), ("SRC_IP", "21.0.0.0/8")] ]
+        for rule in rules:
+           fvs  = swsscommon.FieldValuePairs(rule)
+           subnet_mask_rules += 1
+           tbl.set( "test_subnet|acl_test_rule%s" % subnet_mask_rules, fvs )
+
+        time.sleep(1)
+
+        atbl = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY")
+        keys = atbl.getKeys()
+
+        acl_entry = [k for k in keys if k not in dvs.asicdb.default_acl_entries]
+        assert len(acl_entry) == subnet_mask_rules
+
+        matched_masks = 0
+        for entry in acl_entry:
+           (status, fvs) = atbl.get(entry)
+           assert status == True
+           assert len(fvs) == 6
+           if ('SAI_ACL_ENTRY_ATTR_FIELD_SRC_IP', '23.103.0.0&mask:255.255.192.0') in fvs:
```

Comment:
there is code redundancy in the block as well.

can you define a function as below as use the function to check all entries in acl_entry?

check_rule_existence(entry, rules):
      for rule in rules:
          check if src_ip in the rule match the entry
          check if action of the rule match the entry
          check if priority of the rule match the entry
          if all match:
              then return the True
    if none of the rule match, return false